KubeArmor/BPF/filelessexec.bpf.c

```diff
@@ -0,0 +1,143 @@
+// +build ignore
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright 2023 Authors of KubeArmor */
+
+#include "shared.h"
+
+
+struct {
+  __uint(type, BPF_MAP_TYPE_RINGBUF);
+  __uint(max_entries, 1 << 24);
+} events SEC(".maps");
+
+
+// Force emitting struct mmap_event into the ELF.
+const event *unused __attribute__((unused));
+
+struct preset_map fileless_exec_preset_containers SEC(".maps");
+
+#define MEMFD "memfd:"
+
+static __always_inline int is_memfd(char *name) {
+  char memfd[] = MEMFD;
+  int i = 0;
+  while (i < sizeof(MEMFD) - 1 && name[i] != '\0' && name[i] == memfd[i]) {
+    i++;
+  }
+
+  if (i == sizeof(MEMFD) - 1) {
+    return 1;
+  }
+
+  return 0;
+}
+
+#define RUN_SHM "/run/shm/"
+
+static __always_inline int is_run_shm(char *name) {
+  char run_shm[] = RUN_SHM;
+  int i = 0;
+  while (i < sizeof(RUN_SHM) - 1 && name[i] != '\0' && name[i] == run_shm[i]) {
+    i++;
+  }
+
+  if (i == sizeof(RUN_SHM) - 1) {
+    return 1;
+  }
+
+  return 0;
+}
+
+#define DEV_SHM "/dev/shm/"
+
+static __always_inline int is_dev_shm(char *name) {
```

Comment:
this block of code is being repeated at various places can we do something like this instead? 
```suggestion
static __always_inline int is_dev_shm(char *name , char *is_path) {
```